### PR TITLE
New package: crystal-1.0.0

### PR DESCRIPTION
**Pull request by mipmip**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Extra info

- upgrade to llvm11
- use openssl
- re-enabled do_check()

#### Archs

- I built this PR locally for these architectures with -Q:
  - [x] i686-glibc with -Q
  - [x] x86_64-glibc with -Q
  - [x] x86_64-musl with -Q

Were not able to test the ARM platforms

### CODE REVIEW

**Comment by ericonr** [comment]
I think the commit should be `New package: crystal-1.0.0`? The package was removed and you're reintroducing it.

File: srcpkgs/crystal/template [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+if [ "$build_option_binary_bootstrap" ]; then
+	case "$XBPS_MACHINE" in
+	x86_64)
+		distfiles+=" https://github.com/crystal-lang/crystal/releases/download/${_bootstrapversion}/crystal-${_bootstrapversion}-${_bootstraprevision}-linux-x86_64.tar.gz"
+		checksum+=" 00211ca77758e99210ec40b8c5517b086d2ff9909e089400f6d847a95e5689a4"
+		;;
+	x86_64-musl)
+		distfiles+=" https://github.com/crystal-lang/crystal/releases/download/${_bootstrapversion}/crystal-${_bootstrapversion}-${_bootstraprevision}-linux-x86_64.tar.gz"
+		checksum+=" 00211ca77758e99210ec40b8c5517b086d2ff9909e089400f6d847a95e5689a4"
+		;;
```
Comment (ericonr) [inline srcpkgs/crystal/template:41]:
```suggestion
	x86_64*)
		distfiles+=" https://github.com/crystal-lang/crystal/releases/download/${_bootstrapversion}/crystal-${_bootstrapversion}-${_bootstraprevision}-linux-x86_64.tar.gz"
		checksum+=" 00211ca77758e99210ec40b8c5517b086d2ff9909e089400f6d847a95e5689a4"
		;;
```

Comment (mipmip) [reply srcpkgs/crystal/template]: Thanks, for this suggestion. Did not know about these wildcards in a case statement.

**Comment by mfiano** [comment]
I would love to have Crystal as a package in the official repos. Been keeping an eye on this for a while to weigh my move away from Arch :)

**Comment by notramo** [comment]
I tried to build it locally, but it complains about LLVM_CONFIG versions.

**Comment by notramo** [comment]
@mipmip is there any progress on this?

**Comment by mipmip** [comment]
@notramo No sorry, I moved on to NixOS myself.

**Comment by notramo** [comment]
[ GitHub stale bot considered harmful - Drew DeVault's blog](https://drewdevault.com/2021/10/26/stalebot.html)

**Comment by paper42** [comment]
This PR wasn't updated for 2 years and the author moved away from Void. If it wasn't closed by the bot, we would close it manually. If you want to package this, create a new package PR.

**Pull request closed without merging**